packages/tooling/src/test/kotlin/elide/tooling/codecs/maven/RepositorySystemFactory.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2024-2025 Elide Technologies, Inc.
+ *
+ * Licensed under the MIT license (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ *   https://opensource.org/license/mit/
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under the License.
+ */
+@file:Suppress("DEPRECATION")
+
+package elide.tooling.codecs.maven
+
+import elide.runtime.Logging
+import io.micronaut.context.annotation.Factory
+import jakarta.inject.Singleton
+import org.apache.maven.repository.internal.MavenRepositorySystemUtils
+import org.eclipse.aether.DefaultRepositoryCache
+import org.eclipse.aether.DefaultRepositorySystemSession
+import org.eclipse.aether.RepositorySystem
+import org.eclipse.aether.connector.basic.BasicRepositoryConnectorFactory
+import org.eclipse.aether.impl.DefaultServiceLocator
+import org.eclipse.aether.impl.RemoteRepositoryManager
+import org.eclipse.aether.internal.impl.DefaultRepositorySystem
+import org.eclipse.aether.repository.RepositoryPolicy
+import org.eclipse.aether.spi.connector.RepositoryConnectorFactory
+import org.eclipse.aether.spi.connector.transport.TransporterFactory
+import org.eclipse.aether.transport.file.FileTransporterFactory
+
+@Factory
+internal class RepositorySystemFactory {
```

Comment:
This factory is the same as the one used in the `builder` module, it is required because the Maven codec tests require the repository system and associated components to run, but moving the entire Maven resolution code and dependencies to the `tooling` module would be unnecessary, since consumers would only be interested in the POM model at best.

Reply:
This to me indicates that the split between `builder` and `tooling` needs some adjustments, ideally we would have a clean divide between tooling API and implementation, so thats definitely something to work on.